edb/edgeql/compiler/setgen.py

```diff
@@ -133,32 +133,36 @@ def get_set_type(
 
 def new_set_from_set(
         ir_set: irast.Set, *,
-        preserve_scope_ns: bool=False,
+        merge_current_ns: bool=False,
+        path_scope_id: Optional[int | Literal['']]='',
```

Comment:
Not sure I like `''` as a sentinel.  Maybe use an explicit singleton (a-la `NoDefault`)?

Reply:
Sure